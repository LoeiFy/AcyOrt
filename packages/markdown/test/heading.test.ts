import Markdown, { RenderOptions } from '../src'

it('heading', () => {
  const m1 = new Markdown()
  const h0 = '# An h1 header'
  const h1 = '# [An h1 header](#one)'

  expect(m1.render(h0)).toMatchSnapshot()
  expect(m1.render(h1)).toMatchSnapshot()

  const options: RenderOptions = {
    getHeadingId: (t) => `${t.trim().split(' ').join('')}`,
  }
  expect(m1.render(h0, options)).toMatchSnapshot()

  const m2 = new Markdown(options)
  expect(m2.render(h0)).toMatchSnapshot()
})
