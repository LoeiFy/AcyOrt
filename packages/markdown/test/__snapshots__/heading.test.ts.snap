// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heading 1`] = `
"<h1>
  <a href="#An h1 header" id="An h1 header"></a>An h1 header
</h1>"
`;

exports[`heading 2`] = `
"<h1>
  <a href="#one" id="one"></a>An h1 header
</h1>"
`;

exports[`heading 3`] = `
"<h1>
  <a href="#Anh1header" id="Anh1header"></a>An h1 header
</h1>"
`;

exports[`heading 4`] = `
"<h1>
  <a href="#Anh1header" id="Anh1header"></a>An h1 header
</h1>"
`;
